compiler/AST/checkAST.cpp

```diff
@@ -50,6 +50,29 @@ void checkForDuplicateUses()
   }
 }
 
+void checkArgsAndLocals()
+{
+  // Check that each VarSymbol and each ArgSymbol
+  // has a DefExpr that is in the FnSymbol
+  // or a parent of it.
+  forv_Vec(SymExpr, se, gSymExprs)
+  {
+    DefExpr* def = se->var->defPoint;
+    Symbol* defInSym = def->parentSymbol;
+    Symbol* useInSym = se->parentSymbol;
+
+    if (isFnSymbol(defInSym) && isFnSymbol(useInSym)) {
+      if (defInSym->hasFlag(FLAG_MODULE_INIT) /*||
```

Comment:
I like to write it that way because then I have a good place to put the comment "OK, module init functions can define globals". Nonetheless I don't have a strong opinion here. I think I'll keep it as-is since it's made it past testing.